prelude/slang-cuda-prelude.h

```diff
@@ -190,24 +190,12 @@ typedef size_t NonUniformResourceIndex;
 template<typename T, int ROWS, int COLS>
 struct Matrix;
 
-// Boolean vector types should have the same memory layout as bool arrays
-// Using structs with named fields
-struct bool1
-{
-    bool x;
-};
-struct bool2
-{
-    bool x, y;
-};
-struct bool3
-{
-    bool x, y, z;
-};
-struct bool4
-{
-    bool x, y, z, w;
-};
+// Boolean vector types should follow CUDA's builtin vector alignment rules
+// Using typedef to ucharN to match CUDA native vector types alignment
+typedef uchar1 bool1;
```

Comment:
Isn't this going to cause a lot of warnings? Can we just define them as `struct`s with alignment decorations?